src/flinkSql/flinkLanguageClientManager.ts

```diff
@@ -570,17 +570,22 @@ export class FlinkLanguageClientManager implements Disposable {
             lastWebSocketUrl: this.lastWebSocketUrl,
             websocketUrlMatch: url === this.lastWebSocketUrl,
           });
+
+          // Clear any previous diagnostics for the document *before*
+          // reinitializing the language client (which ends
+          // up reassigning `this.lastDocUri` to null momentarily before
+          // we reassign it after successful language client initialization
+          // in a few lines from now).
+          if (this.lastDocUri) {
+            this.clearDiagnostics(this.lastDocUri);
+          }
```

Comment:
Should this just live inside of `cleanupLanguageClient()`?

Reply:
Tried, didn't seem to work as well. Leaving here for 1.5.x branch.